Source/OxyPlot/Annotations/ImageAnnotation.cs

```diff
@@ -120,17 +122,29 @@ public override void Render(IRenderContext rc)
 
             if (double.IsNaN(width) && double.IsNaN(height))
             {
+                if (this.ImageSource == null)
+                {
+                    throw new InvalidOperationException($"{nameof(this.ImageSource)} is null.");
+                }
```

Comment:
Can we move these up to the top of the method and make the exception message more descriptive? (that said, not sure what to suggest, something like "ImageSource must be non-null before rendering" would just be confusing)

Reply:
<img width="617" alt="image" src="https://user-images.githubusercontent.com/3002068/191017545-c7001cf4-10be-49a3-8bac-5cf9a5cb910a.png">
Please check my post below.
Also, I didn't add an exception to the beginning of the method because the ImageSource is not needed if GetVector returns non-NaN X and Y values. But I didn't look below where the RenderingExtensions.DrawImage call happens, which requires a non-null ImageSource value and which hasn't yet been marked with nullable enable

Reply:
Sorry, I didn't realise that none of the render context stuff has been done yet. The image source is essential to the image annotation, and should be required for the call to `RenderingExtensions.DrawImage`, so it's OK to throw under any circumstance. Maybe like `MathRenderingExtensions` it makes sense to annotate `RenderingExtensions` and `IRenderContext` now (should be pretty simple, since there's no state lingering).